pkg/plugins/manager/loader/finder/local.go

```diff
@@ -182,25 +183,42 @@ func (l *Local) getAbsPluginJSONPaths(path string, followDistFolder bool) ([]str
 				return fmt.Errorf("filepath.Walk reported an error for %q: %w", currentPath, err)
 			}
 
-			if fi.Name() == "node_modules" {
+			switch {
+			case fi.Name() == "node_modules":
 				return util.ErrWalkSkipDir
-			}
-
-			if fi.IsDir() {
+			case fi.Name() == "dist" && !followDistFolder:
+				return util.ErrWalkSkipDir
+			case fi.IsDir():
 				return nil
-			}
-
-			if fi.Name() != "plugin.json" {
+			case fi.Name() != "plugin.json":
 				return nil
 			}
 
-			pluginJSONPaths = append(pluginJSONPaths, currentPath)
+			pluginJSONPaths[currentPath] = struct{}{}
 			return nil
 		}); err != nil {
 		return []string{}, err
 	}
 
-	return pluginJSONPaths, nil
+	// For external plugins, if we found a plugin directory that contains a "dist" directory, we want to ignore the
```

Comment:
I figured post-processing here is easier than adding more logic to the `util` walk function. Considering we are the only consumers of that `Walk` function, I'd like to bring that within the plugins package instead. I'm of the opinion that it's easier to reason about a simpler walk function and some post walk logic to filter paths, than to lump everything within the walk function. But maybe it can be manageable if we bring it in house and refactor it 🤷 cc @grafana/plugins-platform-backend

Reply:
Maybe easier to handle in walk function because you're walking top-down so if you find a dist folder just delete whatever you found before this?

Reply:
I probably didn't consider bundled  plugins, just came to my mind. So maybe post processing is easier anyway? Feels like bugs can easily creep up by matching incorrectly on paths :)

Reply:
BTW when I mean the walk function I mean `util.Walk` not the `WalkFunc` that is provided to the function